src/Chainweb/Version.hs

```diff
@@ -350,3 +360,38 @@ class HasChainwebVersion a where
 instance HasChainwebVersion ChainwebVersion where
     _chainwebVersion = id
     {-# INLINE _chainwebVersion #-}
+
+chainIds :: HasChainwebVersion v => v -> HS.HashSet ChainId
+chainIds = graphChainIds . _chainGraph . _chainwebVersion
+{-# INLINE chainIds #-}
+
+mkChainId
+    :: MonadThrow m
+    => HasChainwebVersion v
+    => Integral i
+    => v
+    -> i
+    -> m ChainId
+mkChainId v i = cid
+    <$ checkWebChainId (chainwebVersionGraph $ _chainwebVersion v) cid
+  where
+    cid = unsafeChainId (fromIntegral i)
+{-# INLINE mkChainId #-}
+
+-- | Sometimes, in particular for testing and examples, some fixed chain id is
+-- needed, but it doesn't matter which one. This function provides some valid
+-- chain id.
+--
+someChainId :: HasCallStack => HasChainwebVersion v => v -> ChainId
+someChainId = head . toList . chainIds
+    -- 'head' is guaranteed to succeed because the empty graph isn't valid chain
+    -- graph.
+{-# INLINE someChainId #-}
+
+-- | Uniformily get a random ChainId
+--
+randomChainId :: HasChainwebVersion v => v -> IO ChainId
+randomChainId v = (!!) (toList cs) <$> randomRIO (0, length cs - 1)
```

Comment:
Naughty `O(n)` lookup once we've expanded to 1,000 chains ;)

Reply:
I am afraid that's the best we can to with a `HashSet`. Well, traversing a list of 1000 elements isn't that bad.